core/common/connector-core/src/main/java/org/eclipse/edc/connector/core/LocalPublicKeyDefaultExtension.java

```diff
@@ -0,0 +1,118 @@
+ *
+ */
+
+package org.eclipse.edc.connector.core;
+
+import org.eclipse.edc.connector.core.security.LocalPublicKeyServiceImpl;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.runtime.metamodel.annotation.Provides;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.iam.LocalPublicKeyService;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.spi.security.KeyParserRegistry;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.configuration.Config;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.util.function.Function.identity;
+import static org.eclipse.edc.connector.core.SecurityDefaultServicesExtension.NAME;
+
+@Extension(value = NAME)
+@Provides(LocalPublicKeyService.class)
+public class LocalPublicKeyDefaultExtension implements ServiceExtension {
+
+    public static final String NAME = "Local Public Key Default Extension";
+
+    public static final String EDC_PUBLIC_KEYS_PREFIX = "edc.iam.publickeys";
+
+    public static final String CONFIG_ALIAS = EDC_PUBLIC_KEYS_PREFIX + ".<pkAlias>.";
+
+    @Setting(context = CONFIG_ALIAS, value = "ID of the public key.", required = true)
+    public static final String ID_SUFFIX = "id";
+
+    @Setting(context = CONFIG_ALIAS, value = "Value of the public key. Multiple formats are supported, depending on the KeyParsers registered in the runtime")
+    public static final String VALUE_SUFFIX = "value";
+
+    @Setting(context = CONFIG_ALIAS, value = "Path to a file that holds the public key, e.g. a PEM file. Multiple formats are supported, depending on the KeyParsers registered in the runtime")
+    public static final String PATH_SUFFIX = "path";
+
+    @Inject
+    public KeyParserRegistry keyParserRegistry;
+
+    private Config keysConfiguration;
+
+    private LocalPublicKeyServiceImpl localPublicKeyService;
+    @Inject
+    private Vault vault;
+
+    @Provider(isDefault = true)
+    public LocalPublicKeyService localPublicKeyService() {
+        return localPublicKeyServiceImpl();
+    }
+
+    private LocalPublicKeyServiceImpl localPublicKeyServiceImpl() {
+        if (localPublicKeyService == null) {
+            localPublicKeyService = new LocalPublicKeyServiceImpl(vault, keyParserRegistry);
+        }
+        return localPublicKeyService;
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        keysConfiguration = context.getConfig(EDC_PUBLIC_KEYS_PREFIX);
```

Comment:
I'd have to cache the context anyway in the initialize right?

Reply:
woa, you're right, I was pretty sure that also the prepare phase had the context injected... at this point it could also remain there, but in my opinion I don't see why context shouldn't be passed to all the phases.